tests/unit/secrets_test.yaml

```diff
@@ -0,0 +1,115 @@
+---
+suite: Secret files
+templates:
+  # primary template files
+  - secrets_st2auth.yaml
+  - deployments.yaml
+  - jobs.yaml
+
+  # included templates must also be listed
+  - configmaps_overrides.yaml
+  - configmaps_packs.yaml
+  - configmaps_rbac.yaml
+  - configmaps_st2-conf.yaml
+  - configmaps_st2-urls.yaml
+  - configmaps_st2web.yaml
+  - secrets_datastore_crypto_key.yaml
+  - secrets_ssh.yaml
+  - secrets_st2apikeys.yaml
+  - secrets_st2auth.yaml
+  - secrets_st2chatops.yaml
+
+tests:
+  - it: ST2 Auth Secret include by default
+    template: secrets_st2auth.yaml
+    set:
+      st2: {}
+    release:
+      name: st2ha
+    asserts:
+      - hasDocuments:
+          count: 1
+      - isNotEmpty:
+          path: data.ST2_AUTH_PASSWORD
+        documentIndex: 0
+      - equal:
+          path: data.ST2_AUTH_USERNAME
+          value: c3QyYWRtaW4=
+        documentIndex: 0
+
+  - it: ST2 Auth Secret set custom username and password
+    template: secrets_st2auth.yaml
+    set:
+      st2:
+        username: example
+        password: badPassword
+    release:
+      name: st2ha
+    asserts:
+      - hasDocuments:
+          count: 1
+      - equal:
+          path: data.ST2_AUTH_USERNAME
+          value: "ZXhhbXBsZQ==" # Base64 encoded value
+        documentIndex: 0
+      - equal:
+          path: data.ST2_AUTH_PASSWORD
+          value: "YmFkUGFzc3dvcmQ="  # Base64 encoded value
+        documentIndex: 0
+
+  - it: ST2 Auth Secret disable generation
+    template: secrets_st2auth.yaml
+    set:
+      st2:
+        existingAuthSecret: "hello-world"
+    release:
+      name: st2ha
+    asserts:
+      - hasDocuments:
+          count: 0
+
+  - it: ST2 Auth Secret custom secret Name
+    template: deployments.yaml
+    set:
+      st2:
+        existingAuthSecret: "hello-world"
+      st2chatops:
+        enabled: true
+    release:
+      name: st2ha
+    asserts:
+      - hasDocuments:
+          count: 14
+      - equal:
+          path: spec.template.spec.initContainers[2].envFrom[0].secretRef.name
+          value: "hello-world"
+        documentIndex: 0
```

Comment:
Could you add a comment on the `documentIndex` for deployments and jobs to say which deployment/job this test is for?


Aside: I really wish we could use something other than index to select documents (and containers, ...) With helm-unittest. But oh well.

Reply:
Resolved

Reply:
oh. Sorry. I was not very specific. I was thinking of something more terse:
```
documentIndex: 0 # st2auth
```

If you look in the other test files, I've done something like that. Is that helpful or confusing for you? If adding this comment is confusing, then I can change how I do it in the other files for clarity.

Reply:
Done